testing/integration/utils.py

```
@@ -138,6 +138,52 @@ def ping_all(self):
+    # Base ports for testing/integration: 7000 + worker_idx * 50
+    # Client ports: [7000, 11999] (stride = 50 ports per worker, up to 100 workers)
+    # Cluster bus:  [17000, 21999]
+    # Coordinator:  [27294, 32294]
+    base = 7000 + (worker_idx % 100) * 50
+    return [base + i * 2 for i in range(num_nodes)]
+
+
+def get_worker_tmpdir(base_dir: str | None = None) -> str:
+    """Return isolated temporary directory for the current worker process."""
+    if not base_dir:
+        base_dir = os.environ.get("TEST_TMPDIR", "/tmp")
```

Ruff (0.16.3)

[error] Probable insecure usage of temporary file or directory: "/tmp"

(S108)